lib/checkbufferoverrun.cpp

```diff
@@ -358,7 +358,7 @@ static bool checkMinSizes(const std::vector<Library::ArgumentChecks::MinSize> &m
         case Library::ArgumentChecks::MinSize::ARGVALUE:
             if (Token::Match(argtok, "%num% ,|)")) {
                 const MathLib::bigint sz = MathLib::toLongNumber(argtok->str());
-                if (sz > arraySize)
+                if (sz > static_cast<MathLib::bigint>(arraySize))
```

Comment:
actually, compiler is not smart enough to not treat your code as warning too - by the way, I use clang, provided by Xcode.